### ISSUE TITLE
Stuck at initialization with Monaco.

### ISSUE THREAD

**Issue by hifall** [REditorSupport/languageserver#26]
I am trying to Monaco editor to connect to a node.js process on Ubuntu 16.04, which in turn connects to the R language server via stdio.

Now I am stuck at the initialize step: the client sends the request to the server, but the server never sends any response, and the client errs because it fails to initialize the server.

The request sent from the client looks like this:

`{"jsonrpc":"2.0","id":0,"method":"initialize","params":{"rootPath":null,"rootUri":"/home","capabilities":{"workspace":{"applyEdit":true,"workspaceEdit":{"documentChanges":true},"didChangeConfiguration":{"dynamicRegistration":false},"didChangeWatchedFiles":{"dynamicRegistration":false},"symbol":{"dynamicRegistration":true},"executeCommand":{"dynamicRegistration":true}},"textDocument":{"synchronization":{"dynamicRegistration":true},"completion":{"completionItem":{"snippetSupport":true},"dynamicRegistration":true},"hover":{"dynamicRegistration":true},"signatureHelp":{"dynamicRegistration":true},"references":{"dynamicRegistration":true},"documentHighlight":{"dynamicRegistration":true},"documentSymbol":{"dynamicRegistration":true},"formatting":{"dynamicRegistration":true},"rangeFormatting":{"dynamicRegistration":true},"onTypeFormatting":{"dynamicRegistration":true},"definition":{"dynamicRegistration":true},"codeAction":{"dynamicRegistration":true},"codeLens":{"dynamicRegistration":true},"documentLink":{"dynamicRegistration":true},"rename":{"dynamicRegistration":true}}},"initializationOptions":{"storagePath":"/"},"trace":"off"}}`

The settings used to start the R language server look like this:

```
{
        "uri": "/home/model.r",
        "rootUri": "/home",
        "command": "R",
        "args": [
            '--quiet',
            '--slave',
            '-e',
            "'languageserver::run()'"
        ]
}
```

I have looked into the server console, but I am unable to find any info/error message I can use to go further.

Any help on this issue would be appreciated!

**Comment by randy3k** [comment]
try `languageserver::run(debug=T)` to look for error messages from stderr.

BTW, it is well known that `node.js` has some trouble connecting to the R stdio on Windows. Not sure it is the same problem on your machine. You can consider using the tcp communications. See for example the atom client https://github.com/REditorSupport/atom-ide-r/blob/master/lib/main.js

**Comment by hifall** [comment]
Thanks @randy3k for your quick reply.

I have tried `languageserver::run(debug=T)`, it doesn't seem to make a difference though -- still no info/debug messages in the server console.

Also, I am on Ubuntu 16.04, instead of Windows.

**Comment by randy3k** [comment]
Did you try without the single quotes
```
        "args": [
            '--quiet',
            '--slave',
            '-e',
            "languageserver::run()"
        ]
```

**Comment by hifall** [comment]
Whoa, that did the trick! Now things start to work really nicely.

Much appreciate your help!

**Comment by randy3k** [comment]
BTW, I have just released v0.2.4 and it includes various improvements.
